### PR TITLE
Feature/filter generated methods

### PR DESCRIPTION
**Pull request by ryan10132**
@nackner for CR

### CODE REVIEW

**Comment by ryan10132** [comment]
Ignore the first commit, I'll rebase off it

File: typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java [hunk 1]
```diff
@@ -50,11 +50,13 @@
     @SuppressWarnings("unchecked")
     private final static List<Class<? extends Annotation>> ANNOTATION_CLASSES = Lists.newArrayList(POST.class, GET.class, DELETE.class, PUT.class, OPTIONS.class);
 
-    public Set<Method> getAllServiceMethods(Class<?> serviceClass) {
+    public Set<Method> getAllServiceMethods(Class<?> serviceClass, TypescriptServiceGeneratorConfiguration settings) {
```
Comment (nackner) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java]: I would just take the methodFilter here, not the full setting object.

Comment (ryan10132) [inline typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java]: done

**Comment by ryan10132** [comment]
Going to merge when the build passes

**Pull request merged**